activesupport/lib/active_support/message_verifier.rb

```diff
@@ -53,7 +53,9 @@ def valid_message?(signed_message)
       return if signed_message.nil? || !signed_message.valid_encoding? || signed_message.blank?
 
       data, digest = signed_message.split("--".freeze)
-      data.present? && digest.present? && ActiveSupport::SecurityUtils.secure_compare(digest, generate_digest(data))
+      data.present? && digest.present? && ActiveSupport::SecurityUtils.fixed_length_secure_compare(digest, generate_digest(data))
+    rescue ArgumentError
+      false
```

Comment:
It does feel weird to do a blanket `ArgumentError` rescue here, though.

Maybe we should leave the `return false unless a.bytesize == b.bytesize` for now.